sqlite-db/migrations/20220526225235_rollover-completed-table.sql

```diff
@@ -0,0 +1,51 @@
+-- Add migration script here
+CREATE TABLE IF NOT EXISTS rollover_completed_event_data (
+    id integer PRIMARY KEY autoincrement,
+    cfd_id integer UNIQUE NOT NULL,
+    event_id integer NOT NULL,
+    settlement_event_id text NOT NULL,
+    refund_timelock text NOT NULL,
+    funding_fee number NOT NULL,
+    rate text NOT NULL,
+    identity text NOT NULL,
+    identity_counterparty text NOT NULL,
+    maker_address text NOT NULL,
+    taker_address text NOT NULL,
+    maker_lock_amount number NOT NULL,
+    taker_lock_amount number NOT NULL,
+    publish_sk text NOT NULL,
+    publish_pk_counterparty text NOT NULL,
+    revocation_secret text NOT NULL,
+    revocation_pk_counterparty text NOT NULL,
+    lock_tx text NOT NULL,
+    lock_tx_descriptor text NOT NULL,
+    commit_tx text NOT NULL,
+    commit_adaptor_signature text NOT NULL,
+    commit_descriptor text NOT NULL,
+    refund_tx text NOT NULL,
+    refund_signature text NOT NULL,
+    FOREIGN KEY (cfd_id) REFERENCES cfds (id),
```

Comment:
We might get around without it but it ensures that this `cfd_id` actually exists. The idea is that there is only 1 active rollover completed table, hence the `unique` criteria. 

It might also simplify look ups for rollover data because if you only care about ther rollover details, you can ignore `event_id` and join it only against `cfds`.

What do you think?

Reply:
hm... it might help us with disassociating it from the event eventually if we ever want to do so. I am fine with keeping it like this, but we have to be aware that this is a pretty hard restriction. This literally means that we can only have one rollover per cfd in the database. 

Given our discussion below (https://github.com/itchysats/itchysats/pull/2140#discussion_r887440179) wouldn't it make more sense to just do an `update` on the table instead of a `delete` and then `insert`?

Reply:
I wanted to have an `update` but could not find an easy way of doing this in a single query because we need to differentiate between `insert` (if no rollover exists so far) and `update` (if this is the second rollover).

Reply:
> hm... it might help us with disassociating it from the event eventually if we ever want to do so. I am fine with keeping it like this, but we have to be aware that this is a pretty hard restriction. This literally means that we can only have one rollover per cfd in the database.

We need to because of our st** event sourcing and aggregates where we load all events and apply them one after each other. Because of this, we need to know which RolloverCompleted event has the actual data attached.  We could do this with a simple `max` or `latest` on a list, but this feels like a hack to me. With a fk relation this is guaranteed that these two rows belong together. 

If we get rid of event sourcing, we can get rid of this fk easily :).